### PR TITLE
Removes omp for ThreadPool in TreeEsemble*

### PR DESCRIPTION
**Pull request by xadupre**

### CODE REVIEW

File: onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h [hunk 1]
```diff
@@ -266,12 +270,22 @@ void TreeEnsembleCommon<ITYPE, OTYPE>::compute_agg(const Tensor* X, Tensor* Z, T
           agg.ProcessTreeNodePrediction1(score, *ProcessTreeNodeLeave(roots_[j], x_data));
       } else {
         std::vector<ScoreValue<OTYPE>> scores_t(n_trees_, {0, 0});
+#if defined(USE_THREADPOOL)
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h:273]: Can you make a runtime check for n_trees > some threshold instead to decide if calling TryBatchParallelFor is needed?

File: onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h [hunk 2]
```diff
@@ -49,14 +53,14 @@ class TreeEnsembleCommon {
...
-  void compute(const Tensor* X, Tensor* Z, Tensor* label) const;
+  void compute(concurrency::ThreadPool* ttp, const Tensor* X, Tensor* Z, Tensor* label) const;
 
  protected:
   TreeNodeElement<OTYPE>* ProcessTreeNodeLeave(
       TreeNodeElement<OTYPE>* root, const ITYPE* x_data) const;
 
   template <typename AGG>
-  void compute_agg(const Tensor* X, Tensor* Z, Tensor* label, const AGG& agg) const;
+  void compute_agg(concurrency::ThreadPool* ttp, const Tensor* X, Tensor* Z, Tensor* label, const AGG& agg) const;
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h:63]: Something to fix later - function names don't conform to coding style.

File: onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h (hunk 1 above)
Comment (xadupre) [reply onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h]: This test is done just above. It kept this switch to be able to compare with the old implementation. I removed it.

File: onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h (hunk 2 above)
Comment (xadupre) [reply onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h:61]: ComputeAgg I assume?

Comment (pranavsharma) [reply onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h:61]: yup

File: onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h [hunk 3]
```diff
@@ -311,24 +315,22 @@ void TreeEnsembleCommon<ITYPE, OTYPE>::compute_agg(const Tensor* X, Tensor* Z, T
...
-            agg.ProcessTreeNodePrediction(private_scores, *ProcessTreeNodeLeave(roots_[j], x_data));
-          }
-
-#ifdef _OPENMP
-#pragma omp critical
-#endif
-          agg.MergePrediction(scores, private_scores);
-        }
-
+        auto nth = n_trees_ * 2 / ttp->NumThreads();
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h:318]: Use ThreadPool::NumThreads(ttp) as ttp can be nullptr.

File: onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h [hunk 4]
```diff
@@ -346,29 +348,31 @@ void TreeEnsembleCommon<ITYPE, OTYPE>::compute_agg(const Tensor* X, Tensor* Z, T
...
-          for (int64_t i = 0; i < N; ++i) {
-            std::fill(scores.begin(), scores.end(), ScoreValue<OTYPE>({0, 0}));
-            for (j = 0; j < roots_.size(); ++j)
-              agg.ProcessTreeNodePrediction(scores, *ProcessTreeNodeLeave(roots_[j], x_data + i * stride));
-            agg.FinalizeScores(scores,
-                               z_data + i * n_targets_or_classes_, -1,
-                               label_data == NULL ? NULL : (label_data + i));
-          }
-        }
+        auto nth = N * 2 / ttp->NumThreads();
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h:351]: Same comment.

File: onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h (hunk 3 above)
Comment (xadupre) [reply onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h]: Done

File: onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h (hunk 2 above)
Comment (xadupre) [reply onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h]: Done

File: onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h (hunk 4 above)
Comment (xadupre) [reply onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h]: Done

**Pull request merged**